botnim/vector_store/vector_score_explainer.py

```diff
@@ -0,0 +1,94 @@
+"""Helper functions for explaining vector similarity scores"""
+
+import numpy as np
+from typing import List, Dict, Any, Optional
+from ..config import get_logger
+
+logger = get_logger(__name__)
+
+def cosine_similarity(vec1: List[float], vec2: List[float]) -> float:
+    """Calculate cosine similarity between two vectors"""
+    vec1_np = np.array(vec1)
+    vec2_np = np.array(vec2)
+    similarity = float(np.dot(vec1_np, vec2_np) / (np.linalg.norm(vec1_np) * np.linalg.norm(vec2_np)))
+    logger.debug(f"Calculated cosine similarity: {similarity:.4f}")
+    return similarity
+
+def explain_vector_scores(query_vector: List[float], doc_vectors: List[Dict[str, Any]]) -> Dict[str, Any]:
+    """
+    Calculate and explain vector similarity scores for document vectors
+    
+    Args:
+        query_vector: Query embedding vector
+        doc_vectors: List of document vectors with source field
+        
+    Returns:
+        Dict with vector similarity explanations
+    """
+    logger.info(f"Processing {len(doc_vectors)} document vectors")
+    logger.debug(f"Vector sources: {[vec.get('source', 'unknown') for vec in doc_vectors]}")
+    
+    vector_scores = []
+    
+    for vec in doc_vectors:
+        source = vec.get("source", "unknown")
+        logger.debug(f"Processing vector from source: {source}")
+        score = cosine_similarity(query_vector, vec["vector"])
+        vector_scores.append({
+            "source": source,
+            "similarity": score,
+            "description": f"Cosine similarity with {source} vector: {score:.4f}"
+        })
+        logger.debug(f"Vector score for {source}: {score:.4f}")
+    
+    # Calculate combined score
+    combined_score = np.mean([s["similarity"] for s in vector_scores])
```

Comment:
Is it necessarily the mean score? Or can we maybe use the max score instead (here and in the query)?

Reply:
@akariv  fixed for the explainer. Correct me if I am wrong, but since the vector search is already using a bool query with "should" clauses for different vector sources (content and description), it means it's already using a max-like behavior